tests/search/apps/test_AppResultItem.py

```diff
@@ -29,23 +39,32 @@ def app_stat_db(self, mocker):
         get_instance.return_value = mock.create_autospec(AppStatDb)
         return get_instance.return_value
 
-    def test_get_name(self, item):
-        assert item.get_name() == 'TestAppResultItem'
+    def test_get_name(self, app1):
+        assert app1.get_name() == 'TrueApp - Full Name'
+
+    def test_get_description(self, app1):
+        assert app1.get_description(Query('q')) == 'Your own yes-man'
+
+    def test_get_icon(self, app1):
+        assert isinstance(app1.get_icon(), GdkPixbuf.Pixbuf)
+        assert app1.get('Icon') == 'dialog-yes'
 
-    def test_get_description(self, item):
-        assert item.get_description(Query('q')) == 'Description of TestAppResultItem'
+    def test_search_score(self, app1):
+        assert app1.search_score("true") > app1.search_score("trivago")
 
-    def test_get_icon(self, item):
-        assert item.get_icon() == 'icon123'
+    def test_search(self, app1, app2):
+        searchresults = AppResultItem.search('false', min_score=0, apps=[app1._app_info, app2._app_info])
+        assert len(searchresults) == 2
+        assert searchresults[0].get_name() == 'FalseApp - Full Name'
 
-    def test_selected_by_default(self, item, query_history):
-        query_history.find.return_value = 'TestAppResultItem'
-        assert item.selected_by_default('q')
+    def test_selected_by_default(self, app1, query_history):
+        query_history.find.return_value = 'TrueApp - Full Name'
+        assert app1.selected_by_default('q')
         query_history.find.assert_called_with('q')
 
-    def test_on_enter(self, item, mocker, query_history, app_stat_db):
+    def test_on_enter(self, app1, mocker, query_history, app_stat_db):
         LaunchAppAction = mocker.patch('ulauncher.search.apps.AppResultItem.LaunchAppAction')
-        assert item.on_enter(Query('query')) is LaunchAppAction.return_value
-        LaunchAppAction.assert_called_with('path/to/desktop_file.desktop')
-        query_history.save_query.assert_called_with('query', 'TestAppResultItem')
-        app_stat_db.inc_count.assert_called_with('path/to/desktop_file.desktop')
+        assert app1.on_enter(Query('query')) is LaunchAppAction.return_value
+        LaunchAppAction.assert_called_with(f'{ENTRIES_DIR}/trueapp.desktop')
```

Comment:
Yeah. Already removed the Ubuntu Xenial build target in the v6 branch, and Bionic was removed accidentally before instead of Xenial, so we can support 3.8 even.

Reply:
This code already ran in pytest, which requires 3.6 anyway :), but I actually want to break 3.5 compatibility immediately with this branch, so no one says it's a regression if we do it later.